src/main/java/makaroshyna/onlinebookstore/repository/book/specification/TitleSpecificationProvider.java

```diff
@@ -0,0 +1,22 @@
+package makaroshyna.onlinebookstore.repository.book.specification;
+
+import static makaroshyna.onlinebookstore.repository.book.BookParameterName.TITLE;
+
+import makaroshyna.onlinebookstore.model.Book;
+import makaroshyna.onlinebookstore.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return TITLE.getName();
+    }
+
+    @Override
+    public Specification<Book> getSpecification(String param) {
+        return (root, query, criteriaBuilder) ->
+                root.get(TITLE.getName()).in(param);
+    }
```

Comment:
you see, it makes sense to filter some books by the author, but I do not see much sense in **filtering** books by their title, people usually **search** by the names of the entity, so let's do some refactors on that, and receive a `searchString` param in the controller, which  we will then pass somewhere down here where we will make `SearchSpecificationProvider` which will be working with SQL `LIKE`, for ex: `root.get(TITLE_KEY).like(searchString)` or maybe ignorecase like or something like that